signer/jws.go

```diff
@@ -0,0 +1,396 @@
+package signer
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+const (
+	MediaTypeJWSJson SignatureMediaType = "application/jose+json"
+)
+
+const (
+	headerKeyExpiry      = "io.cncf.notary.expiry"
+	headerKeySigningTime = "io.cncf.notary.signingTime"
+	headerKeyCrit        = "crit"
+	headerKeyAlg         = "alg"
+	headerKeyCty         = "cty"
+)
+
+// jwsEnvelope represents implements internalSignatureEnvelope interface.
+type jwsEnvelope struct {
+	internalEnv *jwsInternalEnvelope
+}
+
+func newJWSEnvelopeFromBytes(envelopeBytes []byte) (*jwsEnvelope, error) {
+	jwsInternal, err := newJwsInternalEnvelopeFromBytes(envelopeBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &jwsEnvelope{internalEnv: jwsInternal}, nil
+}
+
+func newJWSEnvelope() *jwsEnvelope {
```

Comment:
Declaring a constructor function that just wraps an object declaration is not idiomatic. Just use `new(jwsEnvelope)`.